test.py

```diff
@@ -0,0 +1,147 @@
+
+        if result2 <= result:
+            assert result - result2 <= result2 * 20 / 100
+        else:
+            assert result2 - result <= result * 20 / 100
+
+        if result3 <= result:
+            assert result - result3 <= result3 * 20 / 100
+        else:
+            assert result3 - result <= result * 20 / 100
+
+        assert result >= 0
+        assert result < 100
+
+    if copies == "wholenode":
+        result = wholenodeDiracBenchmark(
+            iterations=iterations, extraIteration=extraIteration
+        )
+        result2 = wholenodeDiracBenchmark(
+            iterations=iterations, extraIteration=extraIteration
+        )
+        result3 = wholenodeDiracBenchmark(
+            iterations=iterations + 1, extraIteration=extraIteration
+        )
+
+        assert result["geometric_mean"] >= 0 and result["geometric_mean"] < 100
+        assert result["arithmetic_mean"] >= 0 and result["arithmetic_mean"] < 100
+        assert result["median"] >= 0 and result["median"] < 100
+
+        for i in result["raw"]:
+            if result2["raw"][k] <= i:
+                assert i - result2["raw"][k] <= result2["raw"][k] * 20 / 100
+            else:
+                assert result2["raw"][k] - i <= i * 20 / 100
+                assert i >= 0
+                assert i < 100
+                k = k + 1
+
+        k = 0
+        for i in result2["raw"]:
+            if result3["raw"][k] <= i:
+                assert i - result3["raw"][k] <= result3["raw"][k] * 20 / 100
+            else:
+                assert result3["raw"][k] - i <= i * 20 / 100
+                assert i >= 0
+                assert i < 100
+                k = k + 1
+
+    if copies == "jobslot":
+        result = jobslotDiracBenchmark(
+            iterations=iterations, extraIteration=extraIteration
+        )
+        result2 = jobslotDiracBenchmark(
+            iterations=iterations, extraIteration=extraIteration
+        )
+        result3 = jobslotDiracBenchmark(
+            iterations=iterations, extraIteration=extraIteration
+        )
+
+        assert result["geometric_mean"] >= 0 and result["geometric_mean"] < 100
+        assert result["arithmetic_mean"] >= 0 and result["arithmetic_mean"] < 100
+        assert result["median"] >= 0 and result["median"] < 100
+
+        for i in result["raw"]:
+            if result2["raw"][k] <= i:
+                assert i - result2["raw"][k] <= result2["raw"][k] * 20 / 100
+            else:
+                assert result2["raw"][k] - i <= i * 20 / 100
+                assert i >= 0
+                assert i < 100
+                k = k + 1
+
+        k = 0
+        for i in result2["raw"]:
+            if result3["raw"][k] <= i:
+                assert i - result3["raw"][k] <= result3["raw"][k] * 20 / 100
+            else:
+                assert result3["raw"][k] - i <= i * 20 / 100
+                assert i >= 0
+                assert i < 100
+                k = k + 1
+    if (
```

Comment:
You can make `if...elif...else` statements instead of multiple `if`, cannot you?
Or maybe also one test function per function to test?

Reply:
I modified it and opted for the if else version